lib/core/types.hpp

```diff
@@ -27,12 +27,13 @@ static constexpr std::uint8_t kNumAccelerometers = 4;
 static constexpr std::uint8_t kNumAxis           = 3;
 static constexpr std::uint8_t kNumEncoders       = 4;
 static constexpr std::uint8_t kNumKeyence        = 2;
+static constexpr std::uint8_t kNumOptical        = 1;
 
 // data format for raw sensor data
 using RawAccelerometerData = std::array<std::array<Float, kNumAxis>, kNumAccelerometers>;
 using AccelerometerData    = std::array<Float, kNumAccelerometers>;
-using EncoderData          = std::array<std::uint32_t, kNumEncoders>;
 using KeyenceData          = std::array<std::uint32_t, kNumKeyence>;
+using OpticalData          = std::array<std::array<Float, 2>, kNumOptical>;
```

Comment:
struct AccelerometerData {
  std::array<Float, kNumAxis> axis_data;
};

struct OpticalData {
  std::array<Float, 2> data;
};

using RawAccelerometerDataArray = std::array<AccelerometerData, kNumAccelerometers>;
using OpticalDataArray = std::array<OpticalData, kNumOptical>;

like this?

Reply:
if arrays of structs are better it would be nicer as
```c++
struct AccelerometerData {
    Float x;
    Float y;
    Float z;
};
```
but tbh it doesn't need to change unless passing the structs around is tidier